### PR TITLE
ROCM HAL Maintenance

### PR DESCRIPTION
**Pull request by raikonenfnu**
-Moving Status/StatusOr to iree/base/status_cc.h.
-IWYU related changes
-Add option to specify where AMD bitcode is

### CODE REVIEW

File: experimental/rocm/dynamic_symbols.c [hunk 1]
```diff
@@ -29,7 +29,7 @@ static iree_status_t iree_hal_rocm_dynamic_symbols_resolve_all(
         syms->loader_library, kName, (void **)&syms->rocmSymbolName)); \
   }
 #define RC_PFN_STR_DECL(rocmSymbolName, ...) RC_PFN_DECL(rocmSymbolName, ...)
-#include "experimental/rocm/dynamic_symbol_tables.h"
+#include "experimental/rocm/dynamic_symbol_tables.h"  // IWYU pragma: keep
```
Comment (ThomasRaoux) [inline experimental/rocm/dynamic_symbols.c:32]: is the comment needed?

File: experimental/rocm/dynamic_symbols.h [hunk 2]
```diff
@@ -19,14 +19,14 @@ extern "C" {
...
-typedef struct {
+typedef struct iree_hal_rocm_dynamic_symbols_t {
   iree_dynamic_library_t *loader_library;
 
 #define RC_PFN_DECL(rocmSymbolName, ...) \
   hipError_t (*rocmSymbolName)(__VA_ARGS__);
 #define RC_PFN_STR_DECL(rocmSymbolName, ...) \
   const char *(*rocmSymbolName)(__VA_ARGS__);
-#include "experimental/rocm/dynamic_symbol_tables.h"
+#include "experimental/rocm/dynamic_symbol_tables.h"  // IWYU pragma: export
```
Comment (ThomasRaoux) [inline experimental/rocm/dynamic_symbols.h:29]: same question

**Review by ThomasRaoux — APPROVED** [review]

File: experimental/rocm/dynamic_symbols.c (hunk 1 above)
Comment (raikonenfnu) [reply experimental/rocm/dynamic_symbols.c:32]: Following the export tag patch [here](https://github.com/google/iree/commit/a0ee7730417a12475dc14508f58c5a25c8de8294#diff-4705a22fccec27fa54445479effa1c26467ae99d224a2102750d90a1b68bd74c)

Comment (ThomasRaoux) [reply experimental/rocm/dynamic_symbols.c:32]: Ha missed that. Looks good.

**Pull request merged**